twoliter/src/lock.rs

```diff
@@ -459,9 +505,48 @@ impl Lock {
             schema_version: project.schema_version(),
             sdk: LockedImage::new(&image_tool, vendor, sdk).await?,
             kit: locked,
+            vendor_override: Self::resolve_overrides(project).await?,
         })
     }
 
+    async fn resolve_overrides(project: &Project) -> Result<HashMap<String, Vendor>> {
+        // Vendor override should either be a vendor.toml file next to where Twoliter.toml is or multiple files defined in
+        // vendor.d/ folder
+        let dir_path = project.project_dir().join("vendor.d");
+        if dir_path.exists() {
+            let mut overrides = HashMap::new();
+            let mut walker = WalkDir::new(&dir_path);
+            let mut files = Vec::new();
+            while let Some(entry) = walker.next().await {
+                let entry = entry.context("failed to check file in vendor.d")?;
+                if let Some(extension) = entry.path().extension() {
+                    if extension == "toml" {
+                        files.push(entry.path());
+                    }
+                }
+            }
+            // Sort the files for override ordering (WalkDir/read_dir does not gaurantee any ordering)
+            files.sort();
+            for file in files.iter() {
+                let next_body = read_to_string(file).await?;
+                let next: VendorOverrideFile = toml::from_str(next_body.as_str())
+                    .context("failed to deserialize vendor override file")?;
+                for (key, value) in next.vendor.iter() {
+                    overrides.insert(key.clone(), value.clone());
+                }
+            }
+            return Ok(overrides);
+        }
+        let file_path = project.project_dir().join("vendor.toml");
```

Comment:
Convention is for our top-level TOML files to start with an upper-case letter, so `Vendor.toml`.

However, I'm a little put off by how generic that name is. Maybe we could support a `.twoliter` directory with various config files instead - e.g. `vendor.toml` or `override.toml` ?

Reply:
Hrm yeah, maybe for vendor it would be .twoliter/vendor.toml or .twoliter/vendor.d Sean had mentioned that we should support multiple vendor files in a directory in addition to single file